### PR TITLE
Remove a bunch of duplication in our telemetry logging

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Just some general cleanup.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Log/HistogramLogAggregator.cs [hunk 1]
```diff
@@ -102,6 +103,18 @@ private int GetBucket(int value)
 
                 return bucket;
             }
+
+            /// <summary>
+            /// Writes out these statistics to a property bag for sending to telemetry.
+            /// </summary>
+            /// <param name="prefix">The prefix given to any properties written; if you want a period as a separator, it's
+            /// your responsibility to pass that.</param>
+            public void WriteTelemetryPropertiesTo(Dictionary<string, object?> properties, string prefix = "")
```
Comment (dibarbet) [inline src/Workspaces/Core/Portable/Log/HistogramLogAggregator.cs:112]: would not be opposed to making prefix required

File: src/Workspaces/Core/Portable/Log/StatisticResult.cs [hunk 2]
```diff
@@ -86,5 +85,25 @@ public StatisticResult(int max, int min, int? median, int mean, int range, int?
             this.Mode = mode;
             this.Count = count;
         }
+
+        /// <summary>
+        /// Writes out these statistics to a property bag for sending to telemetry.
+        /// </summary>
+        /// <param name="prefix">The prefix given to any properties written; if you want a period as a separator, it's
+        /// your responsibility to pass that.</param>
+        public void WriteTelemetryPropertiesTo(Dictionary<string, object?> properties, string prefix = "")
```
Comment (dibarbet) [inline src/Workspaces/Core/Portable/Log/StatisticResult.cs:94]: same here

**Review by dibarbet — APPROVED** [review]

File: src/Workspaces/Core/Portable/Log/HistogramLogAggregator.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/Log/HistogramLogAggregator.cs:112]: I figured an event being fired that just is reporting a single set of these doesn't really need the prefix, but I guess yeah, you can still be explicit. OK changing.

Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/Log/HistogramLogAggregator.cs]: Changed.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**